StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1130UseLambdaSyntax.cs

```diff
@@ -45,37 +47,104 @@ public override void Initialize(AnalysisContext context)
             context.RegisterSyntaxNodeAction(AnonymousMethodExpressionAction, SyntaxKind.AnonymousMethodExpression);
         }
 
+        /// <summary>
+        /// Gets the delegate parameter list from a method symbol and the argument index.
+        /// </summary>
+        /// <param name="methodSymbol">The symbol containing information about the method invocation.</param>
+        /// <param name="argumentIndex">The index of the argument containing the delegate</param>
+        /// <returns>A parameter list for the delegate parameters.</returns>
+        internal static ParameterListSyntax GetDelegateParameterList(IMethodSymbol methodSymbol, int argumentIndex)
+        {
+            var delegateType = (INamedTypeSymbol)methodSymbol.Parameters[argumentIndex].Type;
+            var delegateParameters = delegateType.DelegateInvokeMethod.Parameters;
+
+            var syntaxParameters = GetSyntaxParametersFromSymbolParameters(delegateParameters);
+
+            return SyntaxFactory.ParameterList(SyntaxFactory.SeparatedList(syntaxParameters));
+        }
+
         private static void HandleAnonymousMethodExpression(SyntaxNodeAnalysisContext context)
         {
+            var diagnosticProperties = ImmutableDictionary.CreateBuilder<string, string>();
+
+            bool reportDiagnostic = true;
             var anonymousMethod = (AnonymousMethodExpressionSyntax)context.Node;
 
-            if (anonymousMethod.Parent.IsKind(SyntaxKind.Argument))
+            switch (anonymousMethod.Parent.Kind())
             {
-                // invocation -> argument list -> argument -> anonymous method
-                if (anonymousMethod?.Parent?.Parent?.Parent is InvocationExpressionSyntax originalInvocationExpression)
+            case SyntaxKind.Argument:
+                reportDiagnostic = HandleMethodInvocation(context.SemanticModel, anonymousMethod, (ArgumentSyntax)anonymousMethod.Parent, diagnosticProperties);
+                break;
+
+            case SyntaxKind.EqualsValueClause:
+                reportDiagnostic = true;
+                break;
+
+            case SyntaxKind.AddAssignmentExpression:
+            case SyntaxKind.SubtractAssignmentExpression:
+                reportDiagnostic = true;
+                break;
+            }
+
+            if (reportDiagnostic)
+            {
+                context.ReportDiagnostic(Diagnostic.Create(Descriptor, anonymousMethod.DelegateKeyword.GetLocation(), diagnosticProperties.ToImmutable()));
+            }
+        }
+
+        private static bool HandleMethodInvocation(SemanticModel semanticModel, AnonymousMethodExpressionSyntax anonymousMethod, ArgumentSyntax argumentSyntax, ImmutableDictionary<string, string>.Builder propertiesBuilder)
```

Comment:
:bulb: It looks like `propertiesBuilder` is no longer used

Reply:
Yes, I've missed that one. Its fixed now.